crates/sui-types/src/object.rs

```diff
@@ -476,40 +476,17 @@ pub enum Owner {
     },
     /// Object is immutable, and hence ownership doesn't matter.
     Immutable,
-    /// Object is sequenced via consensus. Ownership is managed by the configured authenticator.
-    ///
-    /// Note: wondering what happened to `V1`? `Shared` above was the V1 of consensus objects.
-    ConsensusV2 {
+    /// Object is exclusively owned by a single address and sequenced via consensus.
+    ConsensusAddressOwner {
         /// The version at which the object most recently became a consensus object.
         /// This serves the same function as `initial_shared_version`, except it may change
         /// if the object's Owner type changes.
         start_version: SequenceNumber,
-        /// The authentication mode of the object
-        authenticator: Box<Authenticator>,
+        // The owner of the object.
+        owner: SuiAddress,
```

Comment:
I don't want to derail this project any more than I already have so don't read too much into this question....

Are we sure that there is no way to have consensus manage objects that are Address owned without requiring this extra user facing complexity? Like i wonder if there was some change we could make in core to allow for the existing Address owned variant to be able to be used as both an owned or shared input.

Reply:
Also now that this is simplified does Owner get to be copy again for a while? ;P (given the changes we're going to make we likely don't want to make this copy again so read this more as a troll comment)

Reply:
I think we could do what you say if we remove fastpath from Sui. Until that happens, this enum is the only way to distinguish whether the object is fastpath or consensus

Reply:
In addition to what Andrew says, Owner isn't really user-facing for most purposes. InputObjectKind is how transactions refer to inputs: https://github.com/MystenLabs/sui/blob/c19df066c90bef3e8c56bacc91287b4fa61d6b0d/crates/sui-types/src/transaction.rs#L2961

We could potentially allow users to name ConsensusAddressOwner objects using `ImmOrOwnedMoveObject` in cases where they want to commit to exact inputs. If possible, this wouldn't require any changes to the Owner struct. The main question is whether we still need an `initialSharedVersion` in this case, which I'm not entirely sure about.